library/src/main/java/com/xwray/groupie/GroupAdapter.java

```diff
@@ -26,9 +26,8 @@
 
     private AsyncDiffUtil.Callback diffUtilCallbacks = new AsyncDiffUtil.Callback() {
         @Override
-        public void onDispatchResult(@NonNull Collection<? extends Group> newGroups) {
-            groups.clear();
-            groups.addAll(newGroups);
+        public void onDispatchAsyncResult(@NonNull Collection<? extends Group> newGroups) {
```

Comment:
renamed this so it's clear that it only happens as part of the `updateAsync` path